roles/reproducer/tasks/push_code.yml

```diff
@@ -124,8 +124,13 @@
       delegate_to: localhost
       when:
         - item.src is abs or item.src is not match('.*:.*')
-      ansible.builtin.command:  # noqa: command-instead-of-module
-        cmd: "rsync -ar {{ item.src }} zuul@controller-0:{{ item.dest }}"
+      ansible.posix.synchronize:
+        src: "{{ item.src }}"
+        dest: "zuul@controller-0:{{ item.dest }}"
+        archive: true
+        recursive: true
+        delete: true
+        use_ssh_args: true
```

Comment:
I'd rather NOT use that `use_ssh_args`:
- controller-0 inventory entry is dynamically injected during the run
- the framework generates specific ssh_config options for all of the managed virtual machines, enabling proxyJump, user and some other things

If really you want to allow overriding the default behavior, you probably should use [rsync_opts](https://docs.ansible.com/ansible/latest/collections/ansible/posix/synchronize_module.html#parameter-rsync_opts) instead, with a default value set to `omit` so that it can be empty/omitted, matching the old behavior.

Reply:
the thing is that without it it might not use certain configurations set to allow ssh connection. generally it is possible to use the rsync opts although I would like to avoid having to duplicate the already existing ssh config from the inventory into another var.